Test failure on s390x: test_nib_tck2trk

While updating the Fedora package to 2.4.1, we received build failures on s390x. 

Hardware information:

```
CPU info:
Architecture:        s390x
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Big Endian
CPU(s):              4
On-line CPU(s) list: 0-3
Thread(s) per core:  1
Core(s) per socket:  1
Socket(s) per book:  1
Book(s) per drawer:  1
Drawer(s):           4
NUMA node(s):        1
Vendor ID:           IBM/S390
Machine type:        2964
CPU dynamic MHz:     5000
CPU static MHz:      5000
BogoMIPS:            3033.00
Hypervisor:          KVM/Linux
Hypervisor vendor:   KVM
Virtualization type: full
Dispatching mode:    horizontal
L1d cache:           128K
L1i cache:           96K
L2d cache:           2048K
L2i cache:           2048K
L3 cache:            65536K
L4 cache:            491520K
NUMA node0 CPU(s):   0-3
Flags:               esan3 zarch stfle msa ldisp eimm dfp edat etf3eh highgprs te vx sie


Memory:
              total        used        free      shared  buff/cache   available
Mem:       10306200      259056     8939776         116     1107368     9935280
Swap:       2097148        2568     2094580


Storage:
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda3       103G  3.2G   95G   4% /
```

Only one test fails:

```
BUILDSTDERR: ======================================================================
BUILDSTDERR: ERROR: nibabel.tests.test_scripts.test_nib_tck2trk
BUILDSTDERR: ----------------------------------------------------------------------
BUILDSTDERR: Traceback (most recent call last):
BUILDSTDERR:   File "/usr/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
BUILDSTDERR:     self.test(*self.arg)
BUILDSTDERR:   File "/builddir/build/BUILD/nibabel-2.4.1/nibabel/tests/test_scripts.py", line 488, in test_nib_tck2trk
BUILDSTDERR:     trk = nib.streamlines.load(standard_trk)
BUILDSTDERR:   File "/builddir/build/BUILD/nibabel-2.4.1/nibabel/streamlines/__init__.py", line 96, in load
BUILDSTDERR:     return tractogram_file.load(fileobj, lazy_load=lazy_load)
BUILDSTDERR:   File "/builddir/build/BUILD/nibabel-2.4.1/nibabel/streamlines/trk.py", line 372, in load
BUILDSTDERR:     arr_seqs = create_arraysequences_from_generator(trk_reader, n=3)
BUILDSTDERR:   File "/builddir/build/BUILD/nibabel-2.4.1/nibabel/streamlines/array_sequence.py", line 387, in create_arraysequences_from_generator
BUILDSTDERR:     for data in gen:
BUILDSTDERR:   File "/builddir/build/BUILD/nibabel-2.4.1/nibabel/streamlines/trk.py", line 683, in _read
BUILDSTDERR:     buffer=f.read(nb_pts * pts_and_scalars_size))
BUILDSTDERR: TypeError: buffer is too small for requested array
BUILDSTDERR: ----------------------------------------------------------------------
```

The complete build log is here:
[build.txt](https://github.com/nipy/nibabel/files/3379111/build.txt)


The build completes fine on x86_64, i686, ppc64le, and armv7hl

<details>
<summary>confirming that the bug is still there with a doctest to follow up</summary>

```
======================================================================
ERROR: nibabel.tests.test_scripts.test_nib_tck2trk
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/yoh/nibabel/nibabel/tests/test_scripts.py", line 488, in test_nib_tck2trk
    trk = nib.streamlines.load(standard_trk)
  File "/home/yoh/nibabel/nibabel/streamlines/__init__.py", line 96, in load
    return tractogram_file.load(fileobj, lazy_load=lazy_load)
  File "/home/yoh/nibabel/nibabel/streamlines/trk.py", line 372, in load
    arr_seqs = create_arraysequences_from_generator(trk_reader, n=3)
  File "/home/yoh/nibabel/nibabel/streamlines/array_sequence.py", line 387, in create_arraysequences_from_generator
    for data in gen:
  File "/home/yoh/nibabel/nibabel/streamlines/trk.py", line 683, in _read
    buffer=f.read(nb_pts * pts_and_scalars_size))
TypeError: buffer is too small for requested array

======================================================================
FAIL: Doctest: nibabel.brikhead.AFNIHeader.__init__
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2224, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for nibabel.brikhead.AFNIHeader.__init__
  File "/home/yoh/nibabel/nibabel/brikhead.py", line 285, in __init__

----------------------------------------------------------------------
File "/home/yoh/nibabel/nibabel/brikhead.py", line 298, in nibabel.brikhead.AFNIHeader.__init__
Failed example:
    header.get_data_dtype()
Expected:
    dtype('int16')
Got:
    dtype('<i2')


----------------------------------------------------------------------
Ran 7835 tests in 90.279s

FAILED (SKIP=62, errors=1, failures=1)
```
</details>

here is some variables

```

PS if interested -- there is a software s390 emulator -- https://hercules-390.github.io/html/ although I have not tried it for ages.  Also I see `qemu-system-misc` package in debian claiming to provide emulators for all kinds of architectures, including s390.  would be cool to try, might do if get bored ;-)
(sid_s390x-dchroot)yoh@zelenka:~/nibabel$ python -m nose -s -v --pdb --pdb-failures /home/yoh/nibabel/nibabel/tests/test_scripts.py:test_nib_tck2trk
nibabel.tests.test_scripts.test_nib_tck2trk ... > /home/yoh/nibabel/nibabel/streamlines/trk.py(683)_read()
-> buffer=f.read(nb_pts * pts_and_scalars_size))
(Pdb) p nb_pts
50331648
(Pdb) p pts_and_scalars_size
12
(Pdb) p f
<nibabel.openers.Opener object at 0x3ffb6749f08>
```

--- a/nibabel/brikhead.py
+++ b/nibabel/brikhead.py
@@ -295,8 +295,8 @@ class AFNIHeader(SpatialHeader):
         --------
         >>> fname = os.path.join(datadir, 'example4d+orig.HEAD')
         >>> header = AFNIHeader(parse_AFNI_header(fname))
-        >>> header.get_data_dtype()
-        dtype('int16')
+        >>> header.get_data_dtype().str
+        '<i2'
         >>> header.get_zooms()
         (3.0, 3.0, 3.0, 3.0)
         >>> header.get_data_shape()

--- a/nibabel/streamlines/trk.py
+++ b/nibabel/streamlines/trk.py
@@ -13,7 +13,7 @@ import nibabel as nib
 
 from nibabel.openers import Opener
 from nibabel.py3k import asstr
-from nibabel.volumeutils import (native_code, swapped_code)
+from nibabel.volumeutils import (native_code, swapped_code, endian_codes)
 from nibabel.orientations import (aff2axcodes, axcodes2ornt)
 
 from .array_sequence import create_arraysequences_from_generator
@@ -266,11 +266,14 @@ class TrkFile(TractogramFile):
             return magic_number == cls.MAGIC_NUMBER
 
     @classmethod
-    def _default_structarr(cls):
+    def _default_structarr(cls, endianness=None):
         """ Return an empty compliant TRK header as numpy structured array
         """
-        # Enforce little-endian byte order for header
-        st_arr = np.zeros((), dtype=header_2_dtype).newbyteorder('<')
+        dt = header_2_dtype
+        if endianness is not None:
+            endianness = endian_codes[endianness]
+            dt = dt.newbyteorder(endianness)
+        st_arr = np.zeros((), dtype=dt)
 
         # Default values
         st_arr[Field.MAGIC_NUMBER] = cls.MAGIC_NUMBER
@@ -284,10 +287,10 @@ class TrkFile(TractogramFile):
         return st_arr
 
     @classmethod
-    def create_empty_header(cls):
+    def create_empty_header(cls, endianness=None):
         """ Return an empty compliant TRK header as dict
         """
-        st_arr = cls._default_structarr()
+        st_arr = cls._default_structarr(endianness)
         return dict(zip(st_arr.dtype.names, st_arr.tolist()))
 
     @classmethod
@@ -396,7 +399,8 @@ class TrkFile(TractogramFile):
             pointing to TRK file (and ready to write from the beginning
             of the TRK header data).
         """
-        header = self._default_structarr()
+        # Enforce little-endian byte order for header
+        header = self._default_structarr(endianness='little')
 
         # Override hdr's fields by those contained in `header`.
         for k, v in self.header.items():

--- a/nibabel/streamlines/trk.py
+++ b/nibabel/streamlines/trk.py
@@ -269,7 +269,8 @@ class TrkFile(TractogramFile):
     def _default_structarr(cls):
         """ Return an empty compliant TRK header as numpy structured array
         """
-        st_arr = np.zeros((), dtype=header_2_dtype)
+        # Enforce little-endian byte order for header
+        st_arr = np.zeros((), dtype=header_2_dtype).newbyteorder('<')
 
         # Default values
         st_arr[Field.MAGIC_NUMBER] = cls.MAGIC_NUMBER
@@ -395,8 +396,7 @@ class TrkFile(TractogramFile):
             pointing to TRK file (and ready to write from the beginning
             of the TRK header data).
         """
-        # Enforce little-endian byte order for header
-        header = self._default_structarr().newbyteorder('<')
+        header = self._default_structarr()
 
         # Override hdr's fields by those contained in `header`.
         for k, v in self.header.items():

